server/sonar-server/src/main/java/org/sonar/server/computation/ComputationContext.java

```diff
@@ -20,31 +20,27 @@
 
 package org.sonar.server.computation;
 
-import com.google.common.annotations.VisibleForTesting;
-import org.sonar.batch.protocol.output.component.ReportComponent;
-import org.sonar.batch.protocol.output.component.ReportComponents;
+import org.sonar.batch.protocol.output.BatchOutputReader;
 import org.sonar.core.component.ComponentDto;
 import org.sonar.core.computation.db.AnalysisReportDto;
 
-import javax.annotation.CheckForNull;
-
-import java.io.File;
 import java.util.Date;
-import java.util.HashMap;
-import java.util.Map;
 
 public class ComputationContext {
 
   private final AnalysisReportDto reportDto;
   private final ComponentDto project;
-  private final File reportDirectory;
-  private Map<Long, ReportComponent> components = new HashMap<>();
-  private Date analysisDate;
+  private final BatchOutputReader reportReader;
+
+  /**
+   * Cache of analysis date as it can be accessed several times
+   */
+  private Date analysisDate = null;
 
-  public ComputationContext(AnalysisReportDto reportDto, ComponentDto project, File reportDir) {
+  public ComputationContext(AnalysisReportDto reportDto, ComponentDto project, BatchOutputReader reportReader) {
```

Comment:
how could it be in json ? Are we talking about another batch report than the computation one ?
I still think that ReportOutputReader or AnalysisReportOutputReader are better names. I don't think it's useful to tell it comes from the batch. WDYT ?